### PR TITLE
Add updated information on domain parameter to OAuth docs

### PR DESCRIPTION
**Pull request by anawishnoff**
We're adding a new query parameter to the OAuth flow

### Motivation

Many partner questions came in regarding the process of using the site parameter, so we are simplifying it and explaining in these docs.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:45]: @bgoldberg122 Just want to confirm that these are correct for the token endpoint

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:31]: @bgoldberg122 isn't this a GET request?

Comment (bgoldberg122) [reply content/en/developers/authorization/oauth2_in_datadog.md:31]: I think it should be GET. According to the docs that we link [here](https://docs.datadoghq.com/developers/authorization/oauth2_endpoints/?tab=authorizationendpoints#request-authorization-from-a-user)

Comment (anawishnoff) [reply content/en/developers/authorization/oauth2_in_datadog.md:31]: yeah - updating

Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:31]:
```suggestion
   - To build the URL for this GET request, use the `domain` query parameter that is provided on the redirect to your `onboarding_url`. 
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:34]:
```suggestion
       - For users authorizing from the US1, Fed1, and EU1 Datadog sites, the URL is: `https://app.<domain>/oauth2/v1/authorize?...`
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:35]:
```suggestion
       - For users authorizing from any other Datadog site, the URL is: `https://<domain>/oauth2/v1/authorize?...`
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (bgoldberg122) [inline content/en/developers/authorization/oauth2_in_datadog.md:33]:
```suggestion
   - The `domain` query parameter provides the [Datadog site][17] that the authorizing user is in, and will be needed to construct the URL for this GET request to the Authorize endpoint: 
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:37]:
```suggestion
<div class="alert alert-info">The `domain` parameter is new as of July 2023. OAuth clients created before then use the `site` parameter instead. The `site` parameter will continue to be supported, but the `domain` parameter is recommended moving forward. </div>
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (bgoldberg122) [inline content/en/developers/authorization/oauth2_in_datadog.md:31]:
```suggestion
   - In order to build the URL for this GET request, use the `domain` query parameter that is provided on the redirect to your `onboarding_url`. 
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:43]:
```suggestion
   - To build the URL for this post request, use the `domain` query parameter that is provided on the redirect to your `redirect_uri`. 
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:45]:
```suggestion
       - For users authorizing from the US1, Fed1, and EU1 Datadog sites, the URL is: `https://api.<domain>/oauth2/v1/token?...`
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:46]:
```suggestion
       - For users authorizing from any other Datadog site, the URL is: `https://<domain>/oauth2/v1/token?...`
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:51]:
```suggestion
    - **Note***: API endpoints are different in each Datadog site. For example, if a user is in the EU region, the events endpoint is `https://api.datadoghq.eu/api/v1/events`, while for users in US1, the events endpoint is `https://api.datadoghq.com/api/v1/events`.
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [inline content/en/developers/authorization/oauth2_in_datadog.md:53]:
```suggestion
    - Some endpoints may also require an API key, which is created in Step 8. 
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (bgoldberg122) [inline content/en/developers/authorization/oauth2_in_datadog.md]: We don't control this flow -- I think the param here might still be `site`

Comment (anawishnoff) [reply content/en/developers/authorization/oauth2_in_datadog.md]: Oh i see - for the token endpoint they'd still use site, but for the authorization endpoint they'd use domain?

Comment (bgoldberg122) [reply content/en/developers/authorization/oauth2_in_datadog.md]: Yeah, I'm not 100% sure how they set the `site` param for the `redirect_uri`. I assume it's the same as how we set it, but maybe they use the correct site param.

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (bgoldberg122) [reply content/en/developers/authorization/oauth2_in_datadog.md]: These are the [docs](https://docs.datadoghq.com/developers/authorization/oauth2_endpoints/?tab=tokenendpoints#exchange-authorization-code-for-access-token). If you want more formal confirmation I would ask AAA

Comment (anawishnoff) [reply content/en/developers/authorization/oauth2_in_datadog.md]: Waiting on confirmation for this

**Comment by anawishnoff** [comment]
@alai97 @bgoldberg122 I think this is finally good to go if y'all wanna take a final look 🙏

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (alai97) [inline content/en/developers/authorization/oauth2_in_datadog.md:52]:
```suggestion
    - **Note***: API endpoints are different in each Datadog site. For example, if a user is in the EU region, the Events endpoint is `https://api.datadoghq.eu/api/v1/events`, while for users in US1, the Events endpoint is `https://api.datadoghq.com/api/v1/events`.
    - You can use the `domain` query parameter directly in these API calls to ensure the correct endpoint is being contacted. For example, to make a call to the Events endpoint, you would build your request URL as `https://api.<domain>/api/v1/events`.
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (alai97) [inline content/en/developers/authorization/oauth2_in_datadog.md:63]:
```suggestion

Users can start the authorization process in Datadog by clicking **Connect Accounts** in the integration tile. When a user clicks Connect Accounts in Datadog, information regarding their [Datadog site][17] is sent on redirect to the `onboarding_url`. The user's Datadog site is required to make API calls on behalf of the user and receive an authorization code. If a user initiates authorization from the _integration's external website_, the user's site information is not provided. 
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (alai97) [inline content/en/developers/authorization/oauth2_in_datadog.md:67]:
```suggestion
Datadog recommends that partners prompt users to initiate authorization from Datadog, rather than from their own platform.
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (alai97) [inline content/en/developers/authorization/oauth2_in_datadog.md:69]:
```suggestion
If you would like to support initiating authorization from a third-party location outside of the Datadog integration tile, ensure that you are able to support users in all Datadog sites, and are willing to continue to support new Datadog sites as they are created. To ensure that users are authorizing in the correct site, direct them to the US1 Datadog site (`app.datadoghq.com`), and from there, they can select their region.
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (alai97) [inline content/en/developers/authorization/oauth2_in_datadog.md:65]:
```suggestion
Additionally, when users initiate authorization from the Datadog integration tile, they are required to have corresponding permissions for all requested scopes. If authorization is initiated from somewhere other than the integration tile, users without all of the required permissions may complete authorization (but are prompted to re-authorize with proper permissions when they return to the integration tile). 
```

**Review by alai97 — COMMENTED** [review]
Looks good, some copy suggestions!

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (alai97) [reply content/en/developers/authorization/oauth2_in_datadog.md:70]:
```suggestion
While Datadog does not recommend supporting initiating authorization from a third-party location anywhere outside of the Datadog integration tile, you must ensure that you are able to support users in all Datadog sites, and are willing to continue to support new Datadog sites as they may be created if you would like to. To ensure that users are authorizing in the correct site, always direct them to the US1 Datadog site (`app.datadoghq.com`), and from there, they can select their region.
```

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [reply content/en/developers/authorization/oauth2_in_datadog.md]: Finalized this with API platform team and updated docs accordingly

**Comment by anawishnoff** [comment]
This PR keeps growing in scope 😅 but I'm also planning on adding a new section about testing. Standby...

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (bgoldberg122) [inline content/en/developers/authorization/oauth2_in_datadog.md:35]:
```suggestion
<div class="alert alert-info">The `domain` parameter was introduced in July 2023, and previous OAuth clients use the `site` parameter. While the `site` parameter is supported, Datadog recommends using the `domain` parameter. </div>
```
Not sure if we want to call out that it's partially supported since it won't work for endpoints like `logs`

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (bgoldberg122) [inline content/en/developers/authorization/oauth2_in_datadog.md:60]: We may want to use the word `domain` instead of `site` everywhere now to avoid confusion with the other url param

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (bgoldberg122) [reply content/en/developers/authorization/oauth2_in_datadog.md]: Did we confirm this?

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [reply content/en/developers/authorization/oauth2_in_datadog.md:35]: I don't think we need to call this out specifically

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [reply content/en/developers/authorization/oauth2_in_datadog.md:60]: I feel like officially it's called the Datadog site, while the domain param is specific to this OAuth implementation

**Review by alai97 — APPROVED** [review]

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (anawishnoff) [reply content/en/developers/authorization/oauth2_in_datadog.md]: confirmed: users should use the site param for the token request, domain param for all other requests

File: content/en/developers/authorization/oauth2_in_datadog.md
Comment (bgoldberg122) [inline content/en/developers/authorization/oauth2_in_datadog.md:37]: Based on our previous conversation, should we say "**Store this value in a secure database or location for later use with additional API endpoints**. "

Comment (anawishnoff) [reply content/en/developers/authorization/oauth2_in_datadog.md:37]: Yeah, I can open a PR to add that

**Pull request merged**
